tests/Application/Actions/Person/GetTest.php

```diff
@@ -253,23 +308,50 @@ private function buildRequestRaw(string $personId): ServerRequestInterface
         return $this->createRequest('GET', '/v1/people/' . $personId);
     }
 
-    private function getStripeClientWithMock(string $mockName): StripeClient
+    /**
+     * @param string $mockName          Main Customer mock name
+     * @param string|null $piMockName   Payment Intent list ["all"] mock name, if needed.
+     */
+    private function getStripeClientWithMock(string $mockName, ?string $piMockName = null): StripeClient
     {
-        $mockResponse = StripeObject::constructFrom(json_decode(
-            file_get_contents(dirname(__DIR__, 3) . '/MockStripeResponses/' . $mockName . '.json'),
-            true,
-            512,
-            JSON_THROW_ON_ERROR,
-        ));
-
         $stripeCustomersProphecy = $this->prophesize(CustomerService::class);
         $stripeCustomersProphecy->retrieve(static::$testPersonStripeCustomerId, ['expand' => ['cash_balance']])
             ->shouldBeCalledOnce()
-            ->willReturn($mockResponse);
+            ->willReturn($this->getStripeObject($mockName));
 
         $stripeClientProphecy = $this->prophesize(StripeClient::class);
         $stripeClientProphecy->customers = $stripeCustomersProphecy->reveal();
 
+        if ($piMockName) {
+            $stripePaymentIntentsProphecy = $this->prophesize(PaymentIntentService::class);
+            // The auto-iterator passes on the `customer` param but also sets its own
+            // limit & other vars – we don't need to test its internal specifics so
+            // we match `Argument::any()`.
+            $stripePaymentIntentsProphecy->all(Argument::any())
+                ->shouldBeCalledOnce()
```

Comment:
I don't really follow the logic for using Argument::any(). We don't need to test the internals of `all` but we do presumably want to test that our SUT is using it correctly.

Also as its a query not a command I think we don't need the `shouldBeCalledOnce:

```suggestion
            $stripePaymentIntentsProphecy = $this->prophesize(PaymentIntentService::class);
            $stripePaymentIntentsProphecy->all(['customer' => self::$testPersonStripeCustomerId])
```

Reply:
I'm pretty sure I started with just that array and it didn't match / use the prophesised call.

There are similar instances in MatchBot which add [just] `'limit' => 100` in cases like this but that didn't work either, and it was late, so I figured `any()` was safe enough for now.

Reply:
I tested this change, the tests do still pass.

Reply:
Oh! Must have had a typo, thanks.